controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -95,6 +95,11 @@ func (a *Adapter) EnsureSnapshotExists() (result controller.OperationResult, err
 		return controller.ContinueProcessing()
 	}
 
+	if _, found := a.pipelineRun.ObjectMeta.Annotations[tekton.PipelineRunChainsSignedAnnotation]; !found {
+		a.logger.Error(err, "Unable to create Snapshot due to missing Chains annotation")
+		return controller.RequeueWithError(err)
```

Comment:
I don't think we want to call `RequeueWithError()` here because it's not an outright error if a pipelineRun is not signed - we just don't want to process such a pipelineRun, so we can log something like `Not processing the pipelineRun because it's not yet signed with Chains` and then call `controller.ContinueProcessing()` instead.

Reply:
ack, updated in most recent change